File: java/PalindromeNumber.java
/*
 * Filename: PalindromeNumber.java
 * Author: Jun Xu
 * Email: junx@bu.edu
 * Created Time: Wed 04 Mar 2015 08:23:59 PM EST
 */

public class Solution {
	public boolean isPalindrome(int x) {
		if (x < 0) {
			return false;
		}

		if (x < 10) {
			return true;
		}

		if (x % 10 == 0) {
			return false;
		}

		int rev = 0;
		int digit;

		while (x > rev) {
			digit = x % 10;
			x = x / 10;
			rev = rev * 10 + digit;
		}

		if (x == rev || x == rev / 10) {
			return true;
		} else {
			return false;
		}
	}
}

File: cpp/palindromeNumber.cpp
/*
 * Filename: palindromeNumber.cpp
 * Author: Jun Xu
 * Email: junx@cs.bu.edu
 * Created Time: Thu 21 Jan 2016 09:35:40 PM EST
 * Description: 
 */

#include <iostream>

using namespace std;

class Solution
{
	public:
		bool isPalindrome(int x)
		{
			if (x < 0)
			{
				return false;
			}

			if (x < 10)
			{
				return true;
			}

			int x2 = x;
			int length = 0;
			int res = 0;
			int digit;

			while (x2)
			{
				x2 /= 10;
				length++;
			}

			for (int i = 0; i < length / 2; i++)
			{
				digit = x % 10;
				x /= 10;
				res = res * 10 + digit;
			}

			if (length % 2)
			{
				return x / 10 == res;
			}
			else
			{
				return x == res;
			}

			return true;
		}
};
